packages/xdl/src/start/ExpoUpdatesManifestHandler.ts

```diff
@@ -29,14 +36,80 @@ function getPlatformFromRequest(req: express.Request | http.IncomingMessage): st
   return String(platform);
 }
 
+async function shouldUseAnonymousManifestAsync(): Promise<boolean> {
+  const currentSession = await UserManager.getSessionAsync();
+  return !currentSession || ConnectionStatus.isOffline();
+}
+
+async function getScopeKeyAndProjectIdForExpoConfigAsync(
+  expoConfig: ExpoConfig
+): Promise<{
+  scopeKey: string;
+  projectId: string | null;
+}> {
+  const shouldUseAnonymousManifest = await shouldUseAnonymousManifestAsync();
+  if (shouldUseAnonymousManifest) {
+    const userAnonymousIdentifier = await UserSettings.getAnonymousIdentifierAsync();
+    return {
+      scopeKey: `@${ANONYMOUS_USERNAME}/${expoConfig.slug}-${userAnonymousIdentifier}`,
+      projectId: null,
+    };
+  }
+
+  const user = await UserManager.ensureLoggedInAsync();
+  const username = await UserManager.getCurrentUsernameAsync();
+  const accountName = expoConfig.owner ?? username;
+  const projectName = expoConfig.slug;
+
+  try {
+    const projects = await ApiV2.clientForUser(user).getAsync('projects', {
+      experienceName: `@${accountName}/${projectName}`,
+    });
+    const project = projects[0];
```

Comment:
Unrelated to this PR but slightly unexpected that a lookup by experience name returns an array

Reply:
The reason it is this way is that we may support filtering by other (non-unique) properties than `experienceName` in the future. This endpoint is designed to be a general REST search method, so returning an array allows us to be fairly future-proof.